src/codec_dav1d.c

```diff
@@ -125,6 +126,23 @@ static avifBool dav1dCodecGetNextImage(struct avifCodec * codec,
         dav1d_data_unref(&dav1dData);
     }
 
+    // Drain all buffered frames in the decoder.
+    Dav1dPicture bufferedFrame;
+    memset(&bufferedFrame, 0, sizeof(Dav1dPicture));
+    do {
+        res = dav1d_get_picture(codec->internal->dav1dContext, &bufferedFrame);
+        if (res < 0) {
+            if (res != DAV1D_ERR(EAGAIN)) {
+                if (gotPicture) {
+                    dav1d_picture_unref(&nextFrame);
```

Comment:
Thank you for looking into this. It is a common API design that a function outputs nothing on failure. This is why I don't call `dav1d_picture_unref(&bufferedFrame)` if `res` is negative. Because of your comment, I checked the documentation. I think we can conclude from the following that a frame is output only when `res` is 0:

```
 * @param out Output frame. The caller assumes ownership of the returned
 *            reference.
 *
 * @return
 *         0: Success, and a frame is returned.
 *  DAV1D_ERR(EAGAIN): Not enough data to output a frame. dav1d_send_data()
 *                     should be called with new input.
 *  Other negative DAV1D_ERR codes: Error during decoding or because of invalid
 *                                  passed-in arguments.
```